src/components/course/EnrollButton.jsx

```diff
@@ -1,117 +1,58 @@
 import React, { useContext, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import qs from 'query-string';
 import { Link, useLocation } from 'react-router-dom';
 import { AppContext } from '@edx/frontend-platform/react';
 import { Button } from '@edx/paragon';
 
 import { UserSubsidyContext } from '../enterprise-user-subsidy';
 import { CourseContext } from './CourseContextProvider';
+import EnrollButtonLabel from './EnrollButtonLabel';
+import EnrollModal from './EnrollModal';
 
-import {
-  ENROLLMENT_FAILED_QUERY_PARAM,
-  COURSE_MODES_MAP,
-} from './data/constants';
+import { COURSE_MODES_MAP } from './data/constants';
+import { useCourseEnrollmentUrl } from './data/hooks';
 import {
   hasCourseStarted,
-  isUserEnrolledInCourse,
+  findUserEnrollmentForCourse,
+  findHighestLevelSeatSku,
+  findOfferForCourse,
 } from './data/utils';
 
-import EnrollButtonLabel from './EnrollButtonLabel';
-import EnrollModal from './EnrollModal';
-
-const getBestCourseMode = (courseModes) => {
-  const {
-    VERIFIED, PROFESSIONAL, NO_ID_PROFESSIONAL, AUDIT,
-  } = COURSE_MODES_MAP;
-  /** Returns the 'highest' course mode available.
-    *  Modes are ranked ['verified', 'professional', 'no-id-professional', 'audit'] */
-  if (courseModes.includes(VERIFIED)) {
-    return VERIFIED;
-  } if (courseModes.includes(PROFESSIONAL)) {
-    return PROFESSIONAL;
-  } if (courseModes.includes(NO_ID_PROFESSIONAL)) {
-    return NO_ID_PROFESSIONAL;
-  }
-  return AUDIT;
-};
-
-const findHighestLevelSeatSku = (seats) => {
-  /** Returns the first seat found from the preferred course mode */
-  if (!seats || seats.length <= 0) {
-    return null;
-  }
-  const courseModes = seats.map(seat => seat.type);
-  const courseMode = getBestCourseMode(courseModes);
-  return seats.find((seat) => seat.type === courseMode).sku;
+const EnrollButtonWrapper = ({
+  as: Component,
```

Comment:
What does this do?

Reply:
`EnrollButtonWrapper` takes in an `as` prop, which is renamed to `Component` and used as the element tag name used for rendering. The enroll button uses Paragon's `Button` component tag as the default, but this can be easily swapped out with `as="a"` or `as={Link}` to change up how the element should be rendered (e.g., if its a native anchor link, react-router's `Link`, or the default `Button`).